### ISSUE TITLE
Add shuffle and split options to CompressedDictionary

### ISSUE THREAD

**Issue by lucadiliello** [iKernels/create-pretraining-datasets-nlp#3]
In place shuffling and an option to split the dictionary in `n` fixed-length dictionaries would be very useful to split in advance the training set.
